src/west/app/main.py

```diff
@@ -384,14 +384,14 @@ def load_extension_specs(self):
             for spec in specs:
                 if spec.name in self.builtins:
                     self.queued_io.append(
-                        lambda cmd: cmd.wrn(
+                        lambda cmd, spec=spec: cmd.wrn(
                             f'ignoring project {spec.project.name} '
                             f'extension command "{spec.name}"; '
                             'this is a built in command'))
                     continue
                 if spec.name in extension_names:
                     self.queued_io.append(
-                        lambda cmd: cmd.wrn(
+                        lambda cmd, spec=spec: cmd.wrn(
```

Comment:
Wait, this one is not just silencing a warning: this looks like an actual bug! So it needs a different commit message.

Unsurprisingly, no one ever complained because this is error handling code which is rarely used and never tested.

So I think the bug should go like this: when west is "ignoring" a project, the error message will always refer to the last `specs[len(spec)-1]` no matter which `spec` was actually involved.

How would one reproduce this wrong error message? I mean interactively at least. If that's complicated then feel free to drop this commit from this cleanup PR so it can be merged quickly.

Reply:
I trust the `spec=spec` fix suggested by https://docs.astral.sh/ruff/rules/function-uses-loop-variable/ is functionally correct, however it's pretty horrible style because it completely obscures the very subtle issue. The two should be different, something like this:

```python
            for spec in specs:
                if spec.name in self.builtins:
                    self.queued_io.append(
                        lambda cmd, spec_const=spec: cmd.wrn(
                            f'ignoring project {spec_const.project.name} '
                            f'extension command "{spec_const.name}"; '
                            'this is a built in command'))
```

Reply:
The issue won't occur once the check has been enabled, I wanted to make as little changes as possible to also preserve git history.

But I can make the changes.